ui/components/component-library/select-wrapper/select-wrapper.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useState, useContext, createContext } from 'react';
+import classnames from 'classnames';
+import type { PolymorphicRef, BoxProps } from '../box';
+import { Box, Popover, PopoverPosition } from '..';
+import {
+  SelectWrapperComponent,
+  SelectWrapperProps,
+} from './select-wrapper.types';
+
+// Should go in the types file
+type SelectContextType = {
+  isOpen: boolean | undefined;
+  onOpenChange: React.Dispatch<React.SetStateAction<boolean | undefined>>; // Double check this is correct type
+  isUncontrolledOpen: boolean;
+  setIsUncontrolledOpen: React.Dispatch<React.SetStateAction<any | null>>;
+  toggleUncontrolledOpen: () => void; // Function to quickly toggle the open state
+  isDisabled: boolean;
+  isMultiSelect: boolean;
+  value: any | null;
+  onValueChange?: any;
+  uncontrolledValue: any | null;
+  setUncontrolledValue: React.Dispatch<React.SetStateAction<any | null>>;
+  defaultValue: any | null;
+  placeholder: any;
+  isDanger: boolean;
+  // onBlur?: React.FocusEventHandler;
+  // onFocus?: React.FocusEventHandler;
+};
+
+export const SelectContext = createContext<SelectContextType | undefined>(
+  undefined,
+);
+
+export const useSelectContext = () => {
+  const context = useContext(SelectContext);
+
+  if (!context) {
+    throw new Error('useSelectContext must be used within a SelectWrapper');
+  }
+
+  return context;
+};
```

Comment:
I think the SelectContext and useSelectContext could go into their own file similarly to the modal context and move it to `select-wrapper.context.ts`